app/javascript/forms/data-driven-form.jsx

```diff
@@ -27,6 +33,7 @@ const MiqFormRenderer = ({ className, ...props }) => (
       'pristine',
       'invalid',
     ]}
+    onStateUpdate={({ pristine }) => setPristine(pristine)}
```

Comment:
This is not going to work if somebody sends `onStateUpdate` (its being used). You should merge this with the callback from props.